### ISSUE TITLE
Contents of routes.txt are loaded into both includeRoutes and excludeRoutes

### ISSUE THREAD

**Issue by Joristdh** [codecentric/spring-boot-admin#1845]
After we updated to spring-boot-admin v2.5.1 we noticed we couldn't directly access our custom toplevel routes anymore. Instead we get a 404 Not Found error. This only happens when directly navigating to this page (via the urlbar) and not when just navigating the UI. Following this, it seems that the contents of _routes.txt_ are ignored and thus the server isn't aware of these routes anymore.

On v2.5.0 this bug does not occur. By looking through the changes between 2.5.0 and 2.5.1 I'm guessing it has something to do with PR #1792 

After looking at _HomepageForwardingMatcher.java_ in the debugger, I noticed that the routes defined in _routes.txt_ were present in both _includedRoutes_ and _excludedRoutes_. This makes _test()_ return false and (I think) results in the 404.

**Comment by Joristdh** [comment]
After digging deeper I found the culprit:
```Java
	@Bean
	public HomepageForwardingFilterConfig homepageForwardingFilterConfig() throws IOException {
		String homepage = this.adminServer.path("/");

		List<String> extensionRoutes = new UiRoutesScanner(this.applicationContext)
				.scan(this.adminUi.getExtensionResourceLocations());
		List<String> routesIncludes = Stream.concat(DEFAULT_UI_ROUTES.stream(), extensionRoutes.stream())
				.map(this.adminServer::path).collect(Collectors.toList());
		List<String> routesExcludes = Stream.concat(DEFAULT_UI_ROUTE_EXCLUDES.stream(), extensionRoutes.stream())
				.map(this.adminServer::path).collect(Collectors.toList());

		return new HomepageForwardingFilterConfig(homepage, routesIncludes, routesExcludes);
	}
```

Here the extensionRoutes is concatenated with into both routesIncludes and routesExcludes.

**Comment by Joristdh** [comment]
Resolved with #1848 in v2.5.3
